src/codelens.ts

```diff
@@ -0,0 +1,31 @@
+import * as vscode from "vscode";
+
+const MarkdownIt = require("markdown-it");
+
+class InlineEvalCodeLensProvider implements vscode.CodeLensProvider {
+  onDidChangeCodeLenses?: vscode.Event<void> | undefined;
+
+  provideCodeLenses(
+    document: vscode.TextDocument,
+    token: vscode.CancellationToken
+  ): vscode.ProviderResult<vscode.CodeLens[]> {
+    const md = new MarkdownIt();
+    const jsFences = md
+      .parse(document.getText())
+      .filter(({ type }: any) => type === "fence")
+      .filter(({ block }: any) => block === true)
+      .filter(({ info }: any) => info === "js");
```

Comment:
Should we also support “javascript”?

Reply:
Should we also support TS, since we already bundle the compiler and it would be a one-line change to compile the code before running it?